lib/active_record/postgres/constraints/types/unique.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module ActiveRecord
```

Comment:
This class duplicates a lot of code in check.rb.  Move that code into a shared module instead of duplicating it.

Also please add tests.